tools/build/Makefile.rules

```diff
@@ -369,6 +374,19 @@ $(SOL_LIB_SO): $(builtin-objs)
 	$(Q)$(LN) -fs $(notdir $(@).$(VERSION)) $(@)
 
 # generators
+define make-oic-gen
+$(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .json) $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .c) $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), -gen.c) $(addprefix $(addprefix $(1), $(notdir $(abspath $(1)))), .h): $(FLOW_OIC_GEN_SCRIPT)
```

Comment:
Why not process these things once, when you are adding the directory to oic-gens and then use those created variables here directly? You are replicating too much stuff below and it gets too error prone.